### PR TITLE
Add how-to guide describing subgraph schema addition

### PR DESCRIPTION
**Pull request by garryod**

### CODE REVIEW

File: docs/how-tos/add-subgraph-schema.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+            uses: actions/upload-artifact@v4.4.0
+            with:
+              name: graphql-schema
+              path: schema.graphql
+    ```
+
+## Check Schema Validity
+
+To check our schema is valid we will use the [Apollo Rover CLI](https://www.apollographql.com/docs/rover/).
+We can retrieve the exisitng schema and configuration by cloning the [`graph-federation` repository](https://github.com/DiamondLightSource/graph-federation/) - in which you will find a `supergraph-config.yaml` file describing the federation and a number of schemas in the `schema/` directory.
```
Comment (daurer) [inline docs/how-tos/add-subgraph-schema.md:49]: mix of you and we, maybe stick to one of them?

File: docs/how-tos/add-subgraph-schema.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+Once cloned, we can add our subgraph definition to the `schema/` directory and a descriptor to the `supergraph-config.yaml`.
```
Comment (daurer) [inline docs/how-tos/add-subgraph-schema.md:51]:
```suggestion
Once cloned, we can add our subgraph definition to the `schema/` directory and add a descriptor to the `supergraph-config.yaml`.
```

**Review by daurer — CHANGES_REQUESTED** [review]
looks good, just a few minor typos and comments. There is a lot of switching between "you" and "we" - would probably make for better reading if just one of them is used consistently...

File: docs/how-tos/add-subgraph-schema.md (hunk 2 above)
Comment (garryod) [reply docs/how-tos/add-subgraph-schema.md:51]: Not sure this suggestion is grammatically correct :sweat_smile:

File: docs/how-tos/add-subgraph-schema.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+To begin, we must generate the subgraph schema of our application.
+How this is performed will differ depending on your chosen language and GraphQL library.
+For instance;
+
+- [`strawberry` (Python)](https://strawberry.rocks/) users can use`strawberry export-schema my_package:schema` as described in the [schema export docs].
+- [`async-graphql` (Rust)](https://crates.io/crates/async-graphql) users should create a command which builds and exports the schema as described on the [SDL export page of the book](https://async-graphql.github.io/async-graphql/en/sdl_export.html).
+
+Your CI job should checkout the source, setup your environment and generate the schema, as shown below.
+It is useful to upload the schema as an artifact to allow download both in future workflow jobs and for review.
```
[the remark this answers is not in the document]
Comment (garryod) [reply docs/how-tos/add-subgraph-schema.md:18]: Original is grammatically correct

**Review by daurer — APPROVED** [review]

**Pull request merged**